src/common/CMakeLists.txt

```diff
@@ -58,13 +61,17 @@ macro( cloudsc_add_common_lib )
         SOURCES
             ${CLOUDSC_COMMON_SOURCES}
             ${COMMON_LIB_SOURCES}
+            $<$<NOT:$<BOOL:${HAVE_FIAT}>>:${CLOUDSC_FIAT_SOURCES}>
             $<${HAVE_HDF5}:module/hdf5_file_mod.F90>
             $<${HAVE_MPI}:module/cloudsc_mpif.F90>
         PUBLIC_LIBS
             $<${HAVE_HDF5}:hdf5::hdf5_fortran>
             $<${HAVE_SERIALBOX}:Serialbox::Serialbox_Fortran>
             $<${HAVE_MPI}:MPI::MPI_Fortran>
             ${COMMON_LIB_LIBS}
+            $<${HAVE_FIELD_API}:fiat>
+            $<${HAVE_FIELD_API}:field_api_${prec}>
+#            $<$<NOT:$<BOOL:${HAVE_FIELD_API}>>:parkind_${prec}>
```

Comment:
And this is redundant in any case because the field_api variant that needs to be linked against field_api does specify this already:
https://github.com/ecmwf-ifs/dwarf-p-cloudsc/blob/4f1f7e5fa6a4b7f76b7c306379d2e3d991a5ae6d/src/common/CMakeLists.txt#L131-L135